### PR TITLE
Added solicitor submit joint application event

### PR DESCRIPTION
**Pull request by nitinprabhu**
https://tools.hmcts.net/jira/browse/NFDIV-1525

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorSubmitJointApplication.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+    private PageBuilder addEventConfig(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+
+        return new PageBuilder(configBuilder.event(SOLICITOR_SUBMIT_JOINT_APPLICATION)
+            .forStates(AwaitingApplicant2Response, Draft)
+            .name("Submit joint application")
+            .description("Submit joint application")
+            .showSummary(false)
+            .endButtonLabel("Submit Application")
+            .grant(CREATE_READ_UPDATE, APPLICANT_2_SOLICITOR, SOLICITOR)
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorSubmitJointApplication.java:61]: is `SOLICITOR` needed?

**Review by linusnorton — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorSubmitJointApplication.java (hunk 1 above)
Comment (nitinprabhu) [reply src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorSubmitJointApplication.java:61]: it's not needed just testing left over thanks

File: src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorSubmitJointApplication.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+        final PageBuilder pageBuilder = addEventConfig(configBuilder);
+
+        pages.forEach(page -> page.addTo(pageBuilder));
+    }
+
+    private PageBuilder addEventConfig(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+
+        return new PageBuilder(configBuilder.event(SOLICITOR_SUBMIT_JOINT_APPLICATION)
+            .forStates(AwaitingApplicant2Response, Draft)
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorSubmitJointApplication.java:55]: this isn't right

Comment (nitinprabhu) [reply src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorSubmitJointApplication.java:55]: yeah local testing left over

**Pull request merged**